ra2ce/runners/direct_analysis_runner.py

```diff
@@ -58,65 +61,124 @@ def can_run(ra2ce_input: ConfigWrapper) -> bool:
         damages_link_based_graph = base_graph_hazard_graph.copy()
 
-        # Step 1: Create a new attribute damage_segments_list for each edge
         for event in events:
-            if damage_curve == analysis.analysis.damage_curve.HZ.name:
-                damage_result_columns = [f'dam_{event}_{damage_curve}']  # there is one damage column
-            elif damage_curve == analysis.analysis.damage_curve.OSD.name:
-                pattern = rf'dam_.*_{event}_representative'
-                damage_result_columns = [col for col in result_segment_based.columns if re.match(pattern, col)]
-                # there are multiple damage columns
-            elif damage_curve == analysis.analysis.damage_curve.MAN.name:
-                pattern = rf'dam_{event}_.*'
-                damage_result_columns = [col for col in result_segment_based.columns if
-                                         re.match(pattern, col)]  # there are multiple damage columns
-            else:
-                raise ValueError(f"damage curve {damage_curve} is invalid")
-            
+            # Step 2: Get the damage_result_columns representing the estimated damages for network segments
+            damage_result_columns = get_damage_result_columns(
+                damage_curve, event, result_segment_based
+            )
+
+            # Step 3: Lookup segment ID list and calculate damage
             for damage_result_column in damage_result_columns:
-                for u, v, key, data in damages_link_based_graph.edges(keys=True, data=True):
-                    damages_link_based_graph[u][v][key]['damage_segments_list'] = []
-                    damages_link_based_graph[u][v][key][damage_result_column] = 0
-
-                # Step 2: Lookup segment_id_list for each edge
-                for u, v, key, data in damages_link_based_graph.edges(keys=True, data=True):
-                    link_damage = data[damage_result_column]
-                    segment_id_list = data[segment_id_column] if isinstance(data[segment_id_column], list) else [
-                        data[segment_id_column]]
-
-                    # Step 3: Read damage for each segment_id & append to damage_segments_list and calculate link_damage
-                    for segment_id in segment_id_list:
-                        segment_damage = result_segment_based.loc[
-                            result_segment_based[segment_id_column] == segment_id, damage_result_column
-                        ].squeeze()
-                        if np.isnan(segment_damage):
-                            segment_damage = 0
-                        data['damage_segments_list'].append(round(segment_damage, 2))
-                        link_damage += round(segment_damage, 2)
-
-                    data[damage_result_column] = round(link_damage, 2)
+                damages_link_based_graph = DirectAnalysisRunner._calculate_link_damage(
```

Comment:
Calling a protected member on a class is an antipattern. It will be solved if you convert `_calculate_link_damage` to an inner function (other remark).